master/internal/db/postgres.go

```diff
@@ -1919,3 +1919,46 @@ WHERE t.id = $1;
 `, id)
 	return errors.Wrapf(err, "error updating best validation for trial %d", id)
 }
+
+// RunResourceAllocationAggregation updates the aggregated resource allocation table.
+func (db *PgDB) RunResourceAllocationAggregation() error {
+	var lastDatePtr *time.Time
+	err := db.sql.QueryRow(
+		`SELECT date_trunc('day', max(date)) FROM resource_aggregates`,
+	).Scan(&lastDatePtr)
+	if err != nil {
+		return errors.Wrap(err, "failed to find last aggregate")
+	}
+
+	var periodStart time.Time
+	if lastDatePtr == nil {
+		var firstDatePtr *time.Time
+		err := db.sql.QueryRow(
+			`SELECT date_trunc('day', min(start_time)) FROM steps`,
+		).Scan(&firstDatePtr)
+		if err != nil {
+			return errors.Wrap(err, "failed to find first step")
+		}
+
+		periodStart = firstDatePtr.UTC()
+	} else {
+		periodStart = lastDatePtr.UTC().AddDate(0, 0, 1)
+	}
+	targetDate := time.Now().UTC().AddDate(0, 0, -1)
+	for ; periodStart.Before(targetDate); periodStart = periodStart.AddDate(0, 0, 1) {
```

Comment:
with this would we end up not aggregating yesterday until tomorrow always? like they'd always have 2 days delay? i'm sorry if i'm reading this wrong, just looks like period start would need to be before yesterday?